### DISCUSSION TITLE
Requests to enable 'Sign in with Google'

### DISCUSSION THREAD

**Discussion by tylerecouture** [bytedeck/bytedeck#1309] [Announcements]
EXPERIMENTAL!

Enabling the "Sign in with Google" feature on individual decks requires us to manually add your deck to our list of authorized URIs at Google.  If you would like this feature for your deck, please post your deck's name in this discussion thread.  After verifying that you are the owner of the deck that you are requesting the feature for, we will enable the feature for you.

A few notes:

**What happens if the gmail address is already verified in a user's profile?**
If the gmail address is already listed in a profile and verified, then attempting to use the Google Sign In will sign them into the account with the already verified gmail address.  This effectively "adds" Google Sign in as a secondary sign in option.

**What happens if the gmail address is in a user's profile but NOT verified?**
Unverified emails are ignored (this prevents unscrupulous users from squatting on another address they don't own).  So if a user has an existing account but has not verified their gmail address, and then they try to use Google Sign In with that gmail address, Bytedeck will prompt them to create a new account.

-Tylere

**Comment by fjbagley** [comment]
Add Studio Tyee, please!

**Reply by tylerecouture** [reply]
This is ready to go, but since you have an established deck, you may want to let students know first, so they don't accidentally create new accounts with their gmail addresses that are disconnected from their existing accounts.  To add Google sign on to existing accounts, they need to set their gmail to their email address in their profile and make sure it's verified.  If they try to sign in with Google before doing this, it will try to create a new account instead.
